tests/integration/individual_response/test_individual_response.py

```diff
@@ -211,6 +231,35 @@ def test_404_individual_response_no_list_items(self):
         # Then I should see the 404 page
         self.assertStatusCode(404)
 
+    def test_500_publish_failed_text(self):
+        publisher = self._application.eq["publisher"]
+        publisher.publish = MagicMock(side_effect=PublicationFailed)
+
+        # Given I add a household member
+        self._add_household_no_primary()
+        self._request_individual_response_by_text()
+        self.assertStatusCode(500)
+        self.assertEqualPageTitle(
+            "Sorry, there was a problem sending the access code - Census 2021"
+        )
+        self.assertInSelectorCSS(
+            self.last_url, "a", href=re.compile("individual-response")
+        )
+
+    def test_500_publish_failed_post(self):
+        publisher = self._application.eq["publisher"]
+        publisher.publish = MagicMock(side_effect=PublicationFailed)
+
+        # Given I add a household member
+        self._add_household_no_primary()
+        self._request_individual_response_by_post()
+        self.assertEqualPageTitle(
+            "Sorry, there was a problem sending the access code - Census 2021"
+        )
+        self.assertInSelectorCSS(
+            self.last_url, "a", href=re.compile("individual-response")
+        )
```

Comment:
Shouldn't the retry url be the same as the last url? That's what I am testing with the above

Reply:
Oh I see now, but that isn't clear. I think it's confusing to search for something using part of the url we're testing for, and also makes the test fragile as it could return another unrelated link. In other templates we use a `data-qa` attribute to select elements directly. As the link is within a string that will be translated, we should consider setting this on the `p` tag (to keep the translatable string as simple as possible), and select the `a` element within that.

Reply:
I'll add an attribute to the `p` element of `data-qa=contact-us`

Reply:
done in dd3818d